### ISSUE TITLE
Does the current syntax make sense?

### ISSUE THREAD

**Issue by ben-eb** [postcss/postcss-use#2]
After fixing a bug with parsing multiple options, are we sure that the current syntax is good enough? Do we want to change it?

Current syntax:

```css
@use autoprefixer (remove: false; browsers: "firefox > 20");
```

Proposed alternatives:

```css
@use autoprefixer (remove: false) and (browsers: "firefox > 20");
```

```css
@use autoprefixer (remove: false, browsers: "firefox > 20");
```

```css
@use autoprefixer {
  remove: false;
  browsers: "firefox > 20";
}
```

**Comment by ai** [comment]
I think semicolon is weird, because semicolon is used in all CSS to separate declarations. So it is some kind of big separator. Not internal separator.

**Comment by jonathantneal** [comment]
I propose you follow the [`@font-face` syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face).

**Comment by TrySound** [comment]
I more like the third example.

**Comment by ai** [comment]
But what we will do, if plugin will accepts a array as options, not a object. 2 and 4 is not ready for arrays.

But it is possible to deny this kind of API. We can add rule to Plugin Guidelines.

**Comment by ai** [comment]
Between 2 and 4, I prefer 4 (font-face syntax), because it is shorter.

**Comment by TrySound** [comment]
comma are combined with parentheses and semicolon with curly brackets

**Comment by TrySound** [comment]
Media queries like looks too large.

**Comment by hagenburger** [comment]
This would be problematic with commas (passing a font-family list could be a use case):

```
@use autoprefixer (remove: false, browsers: "firefox > 20", fonts: "helvetica, arial, sans-serif");
```

The last one feels most natural and allows all the syntax CSS allows without further escaping. I would also try to avoid the quotation marks:

```
@use autoprefixer {
  remove: false;
  browsers: firefox > 20;
  fonts: helvetica, arial, sans-serif;
}
```

**Comment by ai** [comment]
@hagenburger PostCSS has out of box parser for commas with correct support for qutes like in `fonts: "helvetica, arial, sans-serif"`.

**Comment by hagenburger** [comment]
@ai: without quotes it feels more natural.

**Comment by simonsmith** [comment]
Fourth one gets my vote

**Comment by JonathanWolfe** [comment]
I vote for the object notation syntax (and comma seperated as a second-choice).

The `and` variant looks like it should be used for multiple `use`.

**Comment by ben-eb** [comment]
Thanks for your input, everyone. My current thinking is now to accept both the parentheses & the curly braces, so that you can use either/or:

```css
@use autoprefixer (remove: false);
/* Is the same as:
@use autoprefixer {
  remove: false;
}
```

The main reason for this is because the parentheses-style can easily support an array as input:

```css
@use postcss-discard-font-face (['svg', 'ttf']);
```

@hagenburger At this moment in time I am not prepared to support implicit conversion of properties to strings. The reason why is because it is easier for me to just rely on JSON.parse to get the correct value, and in this respect I would rather not re-invent the wheel. Although I am prepared to accept a pull request for this if someone else wants to do that. :smile:

@chriscoyier Would you be open to changing the parentheses style to use commas instead of semi-colons? The people here are in favour of that syntax (`@use plugin (one: "one", two: "two");`). Depending on this I will release either a major or minor release including @jonathantneal's patch.

**Comment by chriscoyier** [comment]
@ben-eb does that mean the parentheses-and-semicolons format will break?  I guess that's not ideal since that's what 1.0.2 supports and is currently live on CodePen. 

Although I tend to think that the curly-brackets-and-semicolons makes the most sense from a CSS perspective and the parens-and-commas makes the most sense from a JSON perspective.

**Comment by ben-eb** [comment]
OK. Let's not break the existing functionality. Will update to add the alternate syntax.

**Comment by ben-eb** [comment]
Released with the alternative syntax as 1.1.0.
